apps/advisor/pkg/app/checktyperegisterer/checktyperegisterer.go

```diff
@@ -105,17 +106,19 @@ func (r *Runner) Run(ctx context.Context) error {
 			},
 		}
 		for i := 0; i < r.retryAttempts; i++ {
-			err := r.createOrUpdate(ctx, obj)
+			err := r.createOrUpdate(ctx, logger, obj)
 			if err != nil {
-				r.log.Error("Error creating check type, retrying", "error", err, "attempt", i+1)
+				logger.Error("Error creating check type, retrying", "error", err, "attempt", i+1)
 				if i == r.retryAttempts-1 {
-					r.log.Error("Unable to register check type")
+					logger.Error("Unable to register check type")
 				} else {
-					time.Sleep(r.retryDelay)
+					// Calculate exponential backoff delay: baseDelay * 2^attempt
+					delay := r.retryDelay * time.Duration(1<<i)
+					time.Sleep(delay)
```

Comment:
Nit: Might be nice to test this - you could extract this to a func that you stub in a test for example

Reply:
it's a bit difficult / cumbersome to test this and if it's mocked we are not getting that much out of it so that's why I skipped it